### ISSUE TITLE
Navigation (TOC) sequence in HTML is decided by the MDX filenames instead of placement of the MDX files

### ISSUE THREAD

**Issue by abhilasha-narendra** [EnterpriseDB/docs#1472]
## Summary
The navigation (TOC) sequence in HTML is decided by the MDX filenames instead of the placement of the MDX files. This makes it difficult to add a new file and place it at a particular place in the guide. If I want to keep the new topic as the 4th section in the guide, then I need to first manually rename the existing 4th topic (which is named 04_<filename>.mdx) as 05_<filename>.mdx, and the rest of the files. Then only I can name my new file as 04_<filename>.mdx to make it appear as the 4th section in the guide. Renaming the files might break the hyperlinks to that topic.

## Expected behavior
The navigation (TOC) sequence should be decided by the placement of the topics, not by the name.

**Comment by josh-heyer** [comment]
You don't have to use file naming to do this anymore; that was an early, and sadly cludgy solution... We now support explicitly specifying the order of the ToC in the index file frontmatter:

```yaml
---
...other frontmatter
navigation:
	- first page filename
	- second page filename
	- ...
	- last page filename
---
```

If you neglect to specify ALL files / subdirectories that should appear in the navigation, the remaining entries will appear as though they had been appended to the list, ordered by filename as they are today.

A real example can be found in https://raw.githubusercontent.com/EnterpriseDB/docs/develop/advocacy_docs/kubernetes/cloud_native_postgresql/index.mdx

**Comment by gsong** [comment]
I'm pretty sure the PDF generation script doesn't respect the `navigation` directive.

**Comment by abhilasha-narendra** [comment]
I tried using the navigation directive in existing docs but it's not working even in the HTML format. Probably it will work only when we remove the numbers prefixed to the filenames, and that would defeat the purpose of using this directive.

**Comment by abhilasha-narendra** [comment]
Apologies for commenting too early. I tried it again and the navigation worked fine, probably it was some alignment issue. Thanks, @josh-heyer for suggesting this resolution.

**Comment by josh-heyer** [comment]
Glad this worked! As George notes, this won't currently affect PDFs; if that's an issue, will need to use a naming convention that allows it for now.

**Comment by drothery-edb** [comment]
Per an earlier conversation, we should figure out how to fix this for PDFs too  so that the PDFs and the HTML have the same structure. Hopefully we would find a solution that doesn't break hyperlinks resulting from renamed files.

**Comment by josh-heyer** [comment]
For compatibility with both PDF and web, can use any naming scheme that allows new files to sort in the proper order and doesn't require renaming existing files. E.g.,

- 01_first.mdx
- 02_second.mdx
- 02a_new_addition.mdx
- 03_third.mdx

This can be done independently of the `navigation` frontmatter, or along with it, as the situation warrants.

**Comment by josh-heyer** [comment]
Fixed by #2415
